### PR TITLE
[Diagnostics-Qol]: Update c_style_for_stmt_removed error string

### LINKED ISSUE #73246: Change "has been removed" to "was removed".
### Motivation

E.g. 

`C-style for statement has been removed in Swift 3`

That implies you're using Swift 3. Nobody is!

### Proposed solution

Either always use "was removed", instead, or automate the conjugation to change.

### Alternatives considered

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by saehejkang**
Changes Made

- Update the `c-style-for-stmt-removed` error string
- Updated appropriate tests in the `/test` directory.

Resolves #73246

### CODE REVIEW

**Comment by saehejkang** [comment]
Just had some clarifications when running tests

1. When do you need to rebuild dependencies when running tests? 
2. This was just updating an error string so does that warrant a rebuild? 
3. What steps should be taken with changes like this (what tests to run)?

**Comment by AnthonyLatsis** [comment]
> When do you need to rebuild dependencies when running tests?

To put is simply, usually, the products you build to compile, debug and verify your changes are the exact same and only products you want to build before running associated tests. If build times are not a problem, `run-test` will build test dependencies for you. Otherwise, you learn them along the way. I prefer using `lit` or skipping the build phase in `run-test` (`--build skip`) because `run-test` is poorly optimized for small subsets of tests — it wastes time rebuilding unrelated stuff. 

> What steps should be taken with changes like this (what tests to run)?

What I do is search for occurrences of the old diagnostic message in the test suite, update them, and make sure those tests pass.

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test Linux

**Comment by AnthonyLatsis** [comment]
Would be nice to have a tag in the commit title too.

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please smoke test

**Review by LucianoPAlmeida — APPROVED** [review]

**Comment by saehejkang** [comment]
@LucianoPAlmeida could you run the ci again?

**Review by AnthonyLatsis — APPROVED** [review]
I prefer a more descriptive commit title than "update", otherwise LGTM.

**Comment by AnthonyLatsis** [comment]
I suggest waiting for an approval from a code owner before rerunning the macOS smoke test. This is just a diagnostic reword and we know from the other successful smoke tests that all the right updates were made, so if the code owners happen to request amendments, we will have run the test for nothing.

**Comment by AnthonyLatsis** [comment]
@bnbarham ping

**Comment by bnbarham** [comment]
LGTM. Thanks @saehejkang! (and @AnthonyLatsis for the ping!)

**Comment by bnbarham** [comment]
@swift-ci please smoke test macOS platform

**Review by xedin — APPROVED** [review]

**Pull request merged**
